src/storage/secondary/column/blob_column_builder.rs

```diff
@@ -80,6 +89,11 @@ impl ColumnBuilder<BlobArray> for BlobColumnBuilder {
                         PlainBlobBlockBuilder::new(self.options.target_block_size - 16),
                     ));
                 }
+                if let Some(to_be_appended) = iter.peek() {
+                    if self.options.record_first_key && to_be_appended.is_some() {
```

Comment:
`None` is also a valid first_key. I think we can decide on encoding of None keys.

Maybe we can use `\0` to represent null keys, and `\1` + BlobEncode to represent non-null keys?

Reply:
Also for the encoding of the key, we can use utilities provided in https://github.com/risinglightdb/risinglight/blob/main/src/storage/secondary/encode.rs

For blob types, simply call `to_be_appended.to_byte_slice().to_vec()`, and store it in `first_key`.

Reply:
We can also `assert!(self.first_key.is_empty())` here, for an extra layer of sanity check.

Reply:
> `None` is also a valid first_key. I think we can decide on encoding of None keys.
> 
> Maybe we can use `\0` to represent null keys, and `\1` + BlobEncode to represent non-null keys?

So we use `\1` for `CharColumnBuilder` too, right?

Reply:
Yes, we use this as a temporary encoding for nullable elements, for both char, blob and primitive types.